### DISCUSSION TITLE
How to select a partial hunk for staging?

### DISCUSSION THREAD

**Discussion by yudistrange** [NeogitOrg/neogit#524] [Q&A] [answered]
Magit allows one to select the individual lines from a hunk and stage them. Neogit mentions that it supports visual mode staging of hunks. Does that mean the same thing? Can I stage partial hunks depending upon the selected line? If so how?

**Comment by CKolkey** [comment]
Yup! Just select the line and hit `s` to stage, or `u` to unstage just the selected lines.

**Reply by yudistrange** [reply]
I tried doing that. But for me the entire hunk is staged.
Please take a look [here](https://drive.google.com/file/d/1oEDMt7KG9N7aAQBQvxubJggfrQkWnK8E/view?usp=drive_link)

I wanted to stage only the `something` function, but the stage operation staged the entire hunk: something + else

**Reply by yudistrange** [reply]
Do I need to configure neogit in some specific way to get this to work?

**Reply by gegoune** [reply, marked as answer]
Use `V` to select lines you want to stage, then press `s`.

**Reply by yudistrange** [reply]
@gegoune Ah! I see. I was using the `v` key instead of the `V`.
That solves it. Thanks!

**Reply by ten3roberts** [reply]
I think that having `v` and `V` behave so inconsistently is a slight bug in itself.

I have also run into this.

Should I create an issue for this?

**Reply by CKolkey** [reply]
It would be pretty trivial to fix - just map `v` to `V` in status.lua. Seems like a reasonable change to test out :)
